compiler/rustc_mir_transform/src/lower_intrinsics.rs

```diff
@@ -35,20 +35,19 @@ impl<'tcx> MirPass<'tcx> for LowerIntrinsics {
                         terminator.kind = TerminatorKind::Goto { target };
                     }
                     sym::forget => {
-                        if let Some(target) = *target {
-                            block.statements.push(Statement {
-                                source_info: terminator.source_info,
-                                kind: StatementKind::Assign(Box::new((
-                                    *destination,
-                                    Rvalue::Use(Operand::Constant(Box::new(ConstOperand {
-                                        span: terminator.source_info.span,
-                                        user_ty: None,
-                                        const_: Const::zero_sized(tcx.types.unit),
-                                    }))),
-                                ))),
-                            });
-                            terminator.kind = TerminatorKind::Goto { target };
-                        }
+                        let target = target.unwrap();
```

Comment:
I'm a bit split on this commit. This could backfire in case we have a pathological input where MIR builder concluded that we don't need a target block, although we need one.

Reply:
The existing code uses a mix of fallible and infallible. Can you see a reason for this split? To me it wasn't clear, it looked arbitrary, so I tried making it all infallible and the test suite passed...

Reply:
The only reason I see is personal taste of the person writing the code.
`target` can only be `None` if the returned type is uninhabited. If the type is known, which happens a lot for intrinsics, we can unwrap without any issue. If the type depends on the user code, we can't.

Reply:
All the intrinsics have a non-`!` return type: `forget`, `{add,sub,mul}_with_overflow`, `size_of`, `min_align_of`, and `discriminant_value`. So I think this change is safe.